### ISSUE TITLE
Sheet: Missing updates when pasting more rows than available in the current sheet

### ISSUE THREAD

**Issue by blutorange** [primefaces-extensions/primefaces-extensions.github.com#588]
## Version

PrimeFaces: 6.2.6, PrimeFaces Extensions: 6.2.7, JSF: mojarra 2.2, Server: jetty/9.4.11.v20180605

## Related

See also https://github.com/primefaces-extensions/primefaces-extensions.github.com/issues/586

## Steps to reproduce

1. Add a `<pe:sheet>` component.
1. Add an `ajax` listener for the `change` event
1. If the data sheet has got, for example, 10 rows, copy 4 rows to the clipboard, click on a cell of the last row, and press ctrl+v to paste those rows.
```xhtml
<pe:sheet>
...
<p:ajax event="change" global="false" listener="#{myBean.onSheetDataChanged}" />
...
<pe:sheet/>
```

On the client side, 3 new rows are created and displayed. However, when we retrieve the updates in the bean via `org.primefaces.extensions.component.sheet.Sheet#getUpdates`, we only get one `SheetUpdate`. All `SheetUpdate`s for cell of newly created rows are missing. This results in inconsistencies between the client and server-side model for me.

As far as I can tell, this happens because a `NullPointerException` was fixed in issue [#586](https://github.com/primefaces-extensions/primefaces-extensions.github.com/issues/586) by [ignoring updates from cells of newly created rows](https://github.com/primefaces-extensions/core/commit/3f5daff0a3b4130475c926bf257e214fb8bf85f2).

**Comment by melloware** [comment]
You are correct.  The model of the backing bean has 10 rows.  So the Sheet Updates get aligned with those backing bean rows.  If 5 new rows are created they don't exist in your backing bean model.  Not sure how i will solve this since I can't create new instances of the objects for you I don't think.  Have to think about this one.

**Comment by blutorange** [comment]
I can create the objects myself, but that requires I know about the additional rows. Perhaps just creating a `SheetUpdate` with a `null` rowkey might work (and `SheetUpdate#isNew`), but I don't know if a null rowkey would cause issues somewhere else. Alternatively, perhaps a `Sheet#getNewlyCreatedRows`?

But take your time, I'm sure you can come up with a better way.

**Comment by melloware** [comment]
Awesome I will close for now!

**Comment by melloware** [comment]
I reopened the issue.  Will investigate when I can.

**Comment by melloware** [comment]
I believe this Stack Overflow user has the same issue: https://stackoverflow.com/questions/60191691/how-to-access-pasted-rows-in-primefaces-extensions-sheet
